CMakeLists.txt

```diff
@@ -162,12 +284,29 @@ add_custom_target(
     linter
     COMMAND ${CMAKE_SOURCE_DIR}/bin/cpplint.sh)
 # Copy the version.h file in before doing anything
-add_custom_target (
-	copy_version_h
-        COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/bin/copy-version.sh"
-)
-add_dependencies(hbaseclient-static copy_version_h)
-add_dependencies(hbaseclient-shared copy_version_h)
+if (NOT BUILD_HBASE)
+	add_custom_target (
+		copy_version_h
+	        COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/bin/copy-version.sh"
+	)
+	
+	add_dependencies(hbaseclient-static copy_version_h)
+	add_dependencies(hbaseclient-shared copy_version_h)
+
+endif(NOT BUILD_HBASE)
```

Comment:
This fails if BUILD_HBASE is false right? (we don't add version.h includes)... I think the code should be structured like this..

set HBASE_SRC_DIR if it is already set by user or set it to the downloaded path in the beginning of the script
Rest of the script shouldn't care about BUILD_HBASE ... they should add dependencies from the set HBASE_SRC_DIR path...

Reply:
Copying the version doesn't really make sense if we rely on either HBASE_HOME or build it within the cmake binary directory. I think the expectation was that it would exist. I don't feel the need to keep the copy version script around. What do you think? 

Thanks for pointing this out.

Reply:
Fair point, agree!